examples/persistent-storage/KeyValueStorageTest.cpp

```diff
@@ -0,0 +1,165 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    Copyright (c) 2019 Google LLC.
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <string>
+
+#include "KeyValueStorage.h"
+#include <support/ErrorStr.h>
+#include <support/logging/CHIPLogging.h>
+
+using namespace chip::Platform::PersistedStorage;
+
+#define RETURN_CHIP_ERROR_IF(cond, chip_error)                                                                                     \
```

Comment:
we have `support/ReturnMacros.h`. Could thse be used or appended to as required?

Reply:
Done